File: app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/action/RecordAction.java
package org.jellyfin.androidtv.ui.playback.overlay.action;

import android.content.Context;
import android.graphics.drawable.Drawable;

import androidx.core.content.ContextCompat;

import org.jellyfin.androidtv.R;
import org.jellyfin.androidtv.ui.playback.overlay.CustomPlaybackTransportControlGlue;

public class RecordAction extends CustomAction {

    public static final int INDEX_INACTIVE = 0;
    public static final int INDEX_RECORDING = 1;

    public RecordAction(Context context, CustomPlaybackTransportControlGlue customPlaybackTransportControlGlue) {
        super(context, customPlaybackTransportControlGlue);
        Drawable recordInactive = ContextCompat.getDrawable(context, R.drawable.ic_record);
        Drawable recordActive = ContextCompat.getDrawable(context, R.drawable.ic_record_red);
        setDrawables(new Drawable[]{recordInactive, recordActive});
    }
}

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/action/RecordAction.kt
package org.jellyfin.androidtv.ui.playback.overlay.action

import android.content.Context
import androidx.core.content.ContextCompat
import org.jellyfin.androidtv.R
import org.jellyfin.androidtv.ui.playback.overlay.CustomPlaybackTransportControlGlue

class RecordAction(
	context: Context,
	customPlaybackTransportControlGlue: CustomPlaybackTransportControlGlue,
) : CustomAction(context, customPlaybackTransportControlGlue) {
	companion object {
		const val INDEX_INACTIVE = 0
		const val INDEX_RECORDING = 1
	}

	init {
		val recordInactive = ContextCompat.getDrawable(context, R.drawable.ic_record)
		val recordActive = ContextCompat.getDrawable(context, R.drawable.ic_record_red)

		setDrawables(arrayOf(recordInactive, recordActive))
	}
}
